File: solutions/2655-rearrange-array-to-maximize-prefix-score/solution.py
class Solution:
    def maxScore(self, nums: List[int]) -> int:
        negative = []
        posValue = 0
        for num in nums:
            if num < 0:
                negative.append(num)
            if num > 0:
                posValue += num   
        if posValue == 0:
            return 0
        if len(negative) == 0:
            return len(nums)
        ans = len(nums) - len(negative)
        negative.sort(reverse=True)
        for num in negative:
            posValue += num
            if posValue > 0:
                ans += 1
            else: break
        return ans

File: solutions/2655-rearrange-array-to-maximize-prefix-score/solution.java
   
class Solution {
   void swap(int[] nums, int i, int j)
   {
       int temp = nums[i];
       nums[i]= nums[j];
       nums[j]=temp;
   }
    public int maxScore(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        int i=0,j=n-1;
        //Reversing the sorted array
        while(i<j)
        {
            swap(nums,i,j);
            i++;
            j--;
        }
       
        int count=0;
        // Making prefixSum long for managing overflow situation
        long prefixSum = 0;
        for(i=0;i<nums.length;i++)
        {
            prefixSum += nums[i];
            if(prefixSum>0)
                count++;

        }
        return count;
        
    }
}
